packages/ra-auth-msal/src/authProvider.ts

```diff
@@ -125,6 +125,7 @@ export const msalAuthProvider = ({
     },
 
     async logout() {
+      await msalInstance.handleRedirectPromise();
```

Comment:
Yes, on every page *load*. I don't think it's necessary here as we load routes (client side only), not pages. It's probably needed in `checkAuth` though as it would execute if you refresh the browser tab for instance

Reply:
Ok, that sounds reasonable, I'll remove await msalInstance.handleRedirectPromise(); from logout().
chekcAuth doesn't use redirect flow directly, it [acquires tokens silently](https://github.com/marmelab/ra-auth-msal/blob/78566ccac0705744a3c371bf20632e3cca9c4dc4/packages/ra-auth-msal/src/authProvider.ts#L150) (neither acquireTokenRedirect  nor acquireTokenPopup). In case of failure, this method calls [login flow](https://github.com/marmelab/ra-auth-msal/blob/78566ccac0705744a3c371bf20632e3cca9c4dc4/packages/ra-auth-msal/src/authProvider.ts#L158), where we have handleRedirectPromise